TLSharp.Core/Network/TcpTransport.cs

```diff
@@ -84,6 +90,42 @@ public void Dispose()
         {
             if (_tcpClient.Connected)
                 _tcpClient.Close();
+        }
+
+        static TcpClient ConnectViaHTTPProxy(string targetHost, int targetPort, WebProxy webProxy)
```

Comment:
It's better to use `Http`, not `HTTP`, to be consistent with the class names in the .NET framework, e.g.: https://msdn.microsoft.com/en-us/library/system.web.httprequest(v=vs.110).aspx